src/Lexer.zig

```diff
@@ -223,14 +223,15 @@ fn lexNumber(self: *Self, literal_value: *LiteralValue) !TokenType {
     var clean_len: usize = 0;
 
     var is_float: bool = false;
+    var is_scientific: bool = false;
     var last_char_is_digit: bool = true;
     var base: u8 = 10; // Default base is decimal.
 
     // Parse until end of number.
     var char = self.source[self.current];
 
     if (!self.isAtEnd() and char == '0') {
-        const next_char = if (self.current + 1 < self.source.len) self.source[self.current + 1] else '?';
+        const next_char = if (!self.isAtEnd()) self.source[self.current + 1] else '?';
```

Comment:
Wouldn't It be better to use `null` instead of random `?` token?
I must have missed It in the original review of this part for some reason.

```suggestion
        const next_char = if (!self.isAtEnd()) self.source[self.current + 1] else null;
```

Reply:
This also requires changing other parts of the code to handle this properly. so committing this suggestion is not a solution